File: LAB3/app/src/main/java/com/example/tipper/MainActivity.java
package com.example.tipper;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private Button bmi, bmr, recipe, quiz;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bmi = (Button) findViewById(R.id.bmi);
        bmi.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openBmiCalculate();
            }
        });

        bmr = (Button) findViewById(R.id.bmr);
        bmr.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openBmrCalculate();
            }
        });

        recipe = (Button) findViewById(R.id.recipe);
        recipe.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                displayRecipe();
            }
        });

        quiz = (Button) findViewById(R.id.quiz);
        quiz.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                displayQuiz();
            }
        });

    }

    private void displayQuiz() {
        Intent intent = new Intent(this, QuizActivity.class);
        startActivity(intent);
    }

    private void displayRecipe() {
        Intent intent = new Intent(this, RecipeActivity.class);
        startActivity(intent);
    }

    private void openBmrCalculate() {
        Intent intent = new Intent(this, BmrActivity.class);
        startActivity(intent);
    }

    public void openBmiCalculate() {
        Intent intent = new Intent(this, BmiActivity.class);
        startActivity(intent);
    }
}

File: LAB4/app/src/main/java/com/example/myapplicationpamo/MainActivity.kt
package com.example.myapplicationpamo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bmi = findViewById<Button>(R.id.bmi)
        bmi.setOnClickListener { displayBmiCalculate() }

        val bmr = findViewById<Button>(R.id.bmr)
        bmr.setOnClickListener { displayBmrCalculate() }

        val recipe = findViewById<Button>(R.id.recipe)
        recipe.setOnClickListener { displayRecipe() }

        val barChart = findViewById<Button>(R.id.barChart)
        barChart.setOnClickListener { displayChart() }

        val quiz = findViewById<Button>(R.id.quiz)
        quiz.setOnClickListener { displayQuiz() }
    }

    private fun displayBmiCalculate() {
        val intent = Intent(this, BmiActivity::class.java)
        startActivity(intent)
    }

    private fun displayBmrCalculate() {
        val intent = Intent(this, BmrActivity::class.java)
        startActivity(intent)
    }

    private fun displayRecipe() {
        val intent = Intent(this, RecipeActivity::class.java)
        startActivity(intent)
    }

    private fun displayChart() {
        val intent = Intent(this, ChartActivity::class.java)
        startActivity(intent)
    }
    private fun displayQuiz() {
        val intent = Intent(this, QuizActivity::class.java)
        startActivity(intent)
    }
}